### ISSUE TITLE
Add emscripten fiber.h api

### ISSUE THREAD

**Issue by unicomp21** [AssemblyScript/assemblyscript#1440]
This might also be needed by:
https://github.com/AssemblyScript/assemblyscript/issues/376

But, I'm thinking having fiber.h will be enough to get started creating an AssemblyScript CSP framework (communicating sequential processes) similar to golang goroutines and channels.

**Comment by dcodeIO** [comment]
I don't see fibers to become relevant any time soon in AS, since JS, and in turn TS, and in turn AS are not designed for in-process threading. The closest thing there is are `WebWorker`s, i.e. cooperative processes with copying in between. If you have something in particular in mind in that regard, please elaborate, but note that fibers are very likely not applicable.

**Comment by MaxGraey** [comment]
I don't think fibers is good approach for our design. Fibers based on stackful coroutines / green threads and required direct access to shadow stack for saving / restoring context (wasm hasn't shadow stack and simulation this only for fibers quite unnecessary). Also fibers event loop (scheduler), threads, atomics, mutex. Which also not always possible due to Wasm MVP hasn't threads and some engines like wasmtime / lucet / wasmer not implement this proposal.

Much better approach is building async / await over generators which based on "asymmetric stackless coroutines". It pretty similar how works javascript's async / await. This could implement similar to [regenerator](https://facebook.github.io/regenerator/) without any requirements for host env and extra proposals.

https://www.youtube.com/watch?v=KUhSjfSbINE

**Comment by MaxGraey** [comment]
Regarding emscripten's `fiber`. It's not "real fibers", just some form of single threaded simulation with stack and threads emulation which required a lot of [javascript glue code](https://github.com/Tandaradei/webgpu-wasm/blob/a3ea4ffd29813fc7c079a5063f52a448a4311848/external/emscripten/src/library_async.js)

**Comment by unicomp21** [comment]
Agree, true async/await would be really nice to have.  But, emscripten fibers are available and ready to use "now".  Why not have both?
